csharp/private/macros/setup_basic_nuget_package.bzl

```diff
@@ -0,0 +1,94 @@
+load("//csharp/private:providers.bzl", "CSharpAssembly")
+load("//csharp/private:rules/imports.bzl", "import_library", "import_multiframework_library")
+
+def _import_dll(dll, has_pdb, imports):
+    path = dll.split("/")
+
+    tfm = path[1]
+
+    # Ignore frameworks we don't support (like net35)
+    if tfm not in CSharpAssembly:
+        return
+
+    lib_name = path[-1].rsplit(".", 1)[0]
+
+    target_name = "%s-%s" % (lib_name, tfm)
+
+    if lib_name not in imports:
+        imports[lib_name] = {tfm: target_name}
+    else:
+        imports[lib_name][tfm] = target_name
+
+    if dll in has_pdb:
+        import_library(
+            name = target_name,
+            target_framework = tfm,
+            dll = dll,
+            pdb = dll[:-3] + "pdb",
+        )
+    else:
+        import_library(
+            name = "%s-%s" % (lib_name, tfm),
+            target_framework = tfm,
+            dll = dll,
+        )
+
+def setup_basic_nuget_package():
+    """This macro gets used to implement the default NuGet BUILD file.
+
+       We are limited by the fact that Bazel does not allow the analysis phase to
+       read the contents of source files, e.g. to correctly configure deps. For
+       more advanced usages a BUILD file will need to be generated outside of
+       Bazel.
+
+       This has to be public so that packages can call it but you probably
+       shouldn't use it directly.
+    """
+    dlls = native.glob(["lib/*/*.dll"])
+    pdbs = native.glob(["lib/*/*.pdb"])
+
+    has_pdb = { (pdb[:-3] + "dll"): 1 for pdb in pdbs }
+
+    # Map from lib name to dict from tfm to target name
+    imports = {}
+
+    # Output import_library rules
+    for dll in dlls:
+        _import_dll(dll, has_pdb, imports)
+
+    # Output import_multiframework_library rules
+    for (name, tfms) in imports.items():
+        import_multiframework_library(
+            name = name,
+            visibility = ["//visibility:public"],
+
+            # I don't know how to do this better.
```

Comment:
I believe this can be replacing using [keyword arguments](https://docs.python.org/3/tutorial/controlflow.html#keyword-arguments). You can see the bazel usage for [miniature](https://docs.bazel.build/versions/master/skylark/tutorial-creating-a-macro.html) has something based on this.

```python
    for tfm in tfms:
        kwargs[tfm.replace(".", "_")] = tfms[tfm]
```

Then `**kwargs` can be used like so:

```python
import_multiframework_library(
    name = name,
    visibility = ["//visibility:public"],
    **kwargs
)
```

Our internal implementation of `copy_files` used it while it was a macro (it no longer does as it doesn't have windows/linux conditionals anymore)

Reply:
Cool, thanks! Let's try it in a follow-up.